src/main/java/com/telekom/m2m/cot/restsdk/users/UsersApi.java

```
@@ -9,226 +9,230 @@
+     * userName, password, firstName, lastName and a tenant.
+     * 
+     * @param userName
+     *            of the user.
+     * @param tenant
+     *            of the tenant where the user resides.
+     * @param firstName
+     *            of the user.
+     * @param lastName
+     *            of the user.
+     * @param password
+     *            of the user.
+     * @return an instance of the user.
+     */
+    public User createUser(String userName, String tenant, String firstName, String lastName, String password) {
+        User user = new User();
+        user.setUserName(userName);
+        user.setFirstName(firstName);
+        user.setLastName(lastName);
+        user.setPassword(password);
+        String json = gson.toJson(user);
+        String id = cloudOfThingsRestClient.doRequestWithIdResponse(json, "user/" + tenant + "/users", CONTENT_TYPE);
+        user.setId(id);
+        return user;
+    }
+
+    /**
+     * A method to create a user in the cloud and return it by providing a
+     * userName, password, firstName, lastName and a tenant.
+     * 
+     * @param userName
+     *            of the user.
+     * @param tenant
+     *            of the tenant where the user resides.
+     * @param firstName
+     *            of the user.
+     * @param lastName
+     *            of the user.
+     * @param password
+     *            of the user.
+     */
+    public void createUserNoReturn(String userName, String tenant, String firstName, String lastName, String password) {
+        User user = new User();
+        user.setUserName(userName);
+        user.setFirstName(firstName);
+        user.setLastName(lastName);
+        user.setPassword(password);
+        String json = gson.toJson(user);
+        String id = cloudOfThingsRestClient.doRequestWithIdResponse(json, "user/" + tenant + "/users", CONTENT_TYPE);
+        user.setId(id);
+    }
+
+    public void deleteUser(User user, String tenant) {
```

These delete-methods do not fit the documentation, because it doesn't mention delete-by-ID at all, just by userName. Maybe that is just bad documentation, because in the examples the values of id and userName are the same, hiding possible errors.
But deviations from the official docs should be documented in the code too.